### ISSUE TITLE
WSIReader Tile Sizes Issue - SVS Format

### ISSUE THREAD

**Issue by TomastpPereira** [TissueImageAnalytics/tiatoolbox#722]
- TIA Toolbox version: 1.4.1
- Python version: 3.11.3
- Operating System: Win 10 x64

### Description

I have been attempting to create WSI tiles using the WSIReader's save_tiles function but have been getting tiles of incorrect shape and content when using WSIs in svs format. 

For a set of WSIs in svs format, the output set of tiles does not match the specified output size at certain objective values. 
By this I mean that I have passed for example, tile_objective_value=5 and tile_read_size=(1024,1024) and the result is a set of patches at (256,256). These tiles also seem to only contain white space. A requested output of (512,512) also instead created (128,128). 
When testing the same issue on sample WSIs provided by Openslide, the issue persists. 

This issue also does not seem to persist when using a WSI in ndpi format. 
The output when using the CMU-1 sample slide from OpenSlide outputs 132 items when using the ndpi format, and only 9 items of incorrect shape when using the svs format. 

I have noted that this seems to occur for all objective values which are below the highest available (20x for the sample).  

### What I Did

```
import openslide
import tiatoolbox
from tiatoolbox.wsicore.wsireader import WSIReader

wsi = WSIReader.open(input_img="C:\\...\\Downloads\\CMU-1.svs")
wsi.save_tiles(output_dir="...", tile_objective_value=5, tile_read_size=(1024,1024))
```

**Comment by shaneahmed** [comment]
Thanks @TomastpPereira Please can you output the type of WSIReader object both for svs and ndpi case? We need to make sure that the objects are created correctly.

**Comment by TomastpPereira** [comment]
Hello @shaneahmed,
For both cases, the type is <class 'tiatoolbox.wsicore.wsireader.OpenSlideWSIReader'>.

In the cases of the svs type, I receive the warning below:
[WARNING] Read: Scale > 1.This means that the desired resolution is higher than the WSI baseline (maximum encoded resolution). Interpolation of read regions may occur.
However, this warning also appears when passing a higher objective value (20), in which case the issue does not occur.  


I have additionally attached the output files for the slide in both types. 
[Output - CMU-1.ndpi.csv](https://github.com/TissueImageAnalytics/tiatoolbox/files/12929895/Output.-.CMU-1.ndpi.csv)
[Output - CMU-1.svs.csv](https://github.com/TissueImageAnalytics/tiatoolbox/files/12929893/Output.-.CMU-1.svs.csv)

**Comment by TomastpPereira** [comment]
Hello,

**Comment by TomastpPereira** [comment]
Hello, apologies for the error in closing the issue. Has there been any progress towards the issue?

Kind Regards, 
Tomas Pereira

**Comment by TomastpPereira** [comment]
Hello, apologies for the error in closing the issue. Has there been any progress towards the issue?

Kind Regards,
Tomas Pereira

**Comment by measty** [comment]
The reason this fails is because of these lines:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/a4463ee41a28a78adb3ba7627bf841278c41ea15/tiatoolbox/wsicore/wsireader.py#L992C1-L998

This only works if the levels are 'all there', ie the info.level_downsamples looks like `[1,2,4,8,16,....]`. In many slides, not all the levels are there, it might just be `[1,4,16,32]` for example. In these cases the code above breaks due to a false assumption.

I've made a PR that I think fixes it in #752
